jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -255,11 +258,67 @@ ecma_op_object_find_own (ecma_object_t *object_p, /**< the object */
     }
   }
 
-  ecma_property_t *property_p = ecma_op_object_get_own_property (object_p, property_name_p);
+  ecma_property_t *property_p = ecma_find_named_property (object_p, property_name_p);
 
   if (property_p == NULL)
   {
-    return ecma_make_simple_value (ECMA_SIMPLE_VALUE_NOT_FOUND);
+    if (type == ECMA_OBJECT_TYPE_STRING)
+    {
+      uint32_t index;
+
+      if (ecma_string_get_array_index (property_name_p, &index))
+      {
+        ecma_value_t *prim_value_p = ecma_get_internal_property (object_p,
+                                                                 ECMA_INTERNAL_PROPERTY_ECMA_VALUE);
+
+        ecma_string_t *prim_value_str_p = ecma_get_string_from_value (*prim_value_p);
+
+        if (index < ecma_string_get_length (prim_value_str_p))
+        {
+          ecma_char_t c = ecma_string_get_char_at_pos (prim_value_str_p, index);
+          return ecma_make_string_value (ecma_new_ecma_string_from_code_unit (c));
+        }
+      }
```

Comment:
Yes, those property names are just regular properties. E.g: 

    var a = new String("x")
    a[0] = 1; // nothing happens
    a[1] = 1; // a[1] is defined with a value of 1